src/main/java/ui/MenuControl.java

```diff
@@ -157,14 +136,46 @@ private void onBoardSaveAs() {
         ui.showMainStage();
         panels.selectFirstPanel();
 
-        if (response.isPresent()) {
-            String boardName = response.get().trim();
-            prefs.addBoard(boardName, panelList);
-            prefs.setLastOpenBoard(boardName);
-            ui.triggerEvent(new BoardSavedEvent());
-            logger.info("New board " + boardName + " saved");
-            ui.updateTitle();
+        if (!response.isPresent()) {
+            return false;
         }
+
+        String boardName = response.get().trim();
+        prefs.addBoard(boardName, panelList);
+        prefs.setLastOpenBoard(boardName);
+        prefs.setLastOpenBoardPanelInfo(panelList);
+        ui.triggerEvent(new BoardSavedEvent());
+        logger.info("New board " + boardName + " saved");
+        ui.updateTitle();
+        return true;
+    }
+
+    /**
+     * Prompts the user for whether to save the current board panel info or not
+     * when user is closing current board (i.e. switching to another board)
+     *
+     * @return true if user choose to save, false if user choose to discard
+     */
+    private boolean promptToSaveCurrentBoard() {
+        List<PanelInfo> currentPanelInfo = panels.getCurrentPanelInfos();
+        List<PanelInfo> lastOpenBoardPanelInfo = prefs.getLastOpenBoardPanelInfo();
+
+        if (currentPanelInfo.equals(lastOpenBoardPanelInfo)) {
+            // No changes to current board, no need to save.
+            return false;
+        }
```

Comment:
This makes the implementation not-as-advertised. If the method is for 'prompting the user', the decision whether to prompt should be taken outside the method. The code should read like
```
if(hasBoardChanged(...)) promptToSaveCurrentBoard()
```